c2corg_ui/static/js/simplesearch.js

```diff
@@ -230,16 +269,16 @@ app.SimpleSearchController.prototype.createAndInitBloodhound_ = function(type) {
       rateLimitWait: 50,
       prepare: (function(query, settings) {
 
-        var url = settings['url'] +
-                '&pl=' + this.gettextCatalog_.currentLanguage;
+        var url = settings['url'] + '&pl=' + this.gettextCatalog_.currentLanguage;
 
         if (this.datasetLimit_) {
-          if (this.datasetLimit_.length === 1) {
-            url = url + '&t=' + this.datasetLimit_;
-          } else {
-            // you receive for ex 'wru' (waypoints, routes, users) -> split into w,r,u
-            this.datasetLimit_ = this.datasetLimit_.split('').join(',');
+          // add the Auth header if searching for users
+          if (this.datasetLimit_.indexOf('u') > -1) {
+            settings['headers'] = {
+              'Authorization': 'JWT token="' + this.auth_.getUserToken() + '"'
```

Comment:
What about using ``this.userData.token`` instead as suggested by @gberaudo ?
What happens if the user is not authenticated? In that case it is not relevant to add an ``Authorization`` header, is it?